editor/src/components/canvas/canvas-strategies/strategies/set-flex-gap-strategy.spec.browser2.tsx

```diff
@@ -0,0 +1,332 @@
+  })
+  it('gap controls are present when flex gap is set on element and element has children', async () => {
+    const editor = await renderTestEditorWithCode(
+      `import * as React from 'react'
+import { Scene, Storyboard } from 'utopia-api'
+
+export var storyboard = (
+  <Storyboard data-uid='0cd'>
+    <div
+      data-testid='${DivTestId}'
+      style={{
+        backgroundColor: '#0091FFAA',
+        position: 'absolute',
+        left: 167,
+        top: 180,
+        width: 557,
+        height: 359,
+        display: 'flex',
+        gap: 42
+      }}
+      data-uid='fac'
+    >
+      <div
+        style={{
+          backgroundColor: '#0091FFAA',
+          width: 102,
+          height: 80,
+          contain: 'layout',
+        }}
+        data-uid='fed'
+      />
+      <div
+        style={{
+          backgroundColor: '#0091FFAA',
+          width: 187,
+          height: 150,
+          contain: 'layout',
+        }}
+        data-uid='a39'
+      />
+    </div>
+  </Storyboard>
+)
+`,
+      'await-first-dom-report',
+    )
+
+    const canvasControlsLayer = editor.renderedDOM.getByTestId(CanvasControlsContainerID)
+    const div = editor.renderedDOM.getByTestId(DivTestId)
+    const divBounds = div.getBoundingClientRect()
+    mouseClickAtPoint(canvasControlsLayer, {
+      x: divBounds.x + 5,
+      y: divBounds.y + 5,
+    })
+
+    const gapControlContainer = editor.renderedDOM.getByTestId(FlexGapControlTestId)
+    const gapControlHandles = editor.renderedDOM.queryAllByTestId(FlexGapControlHandleTestId)
+
+    expect(gapControlContainer).toBeTruthy()
+    expect(gapControlHandles.length).toEqual(1)
+  })
+
+  it('adjusts gap by dragging the handle', async () => {
+    const editor = await renderTestEditorWithCode(
+      testCodeWithGap({ gap: `gap: '42px',`, flexDirection: 'row' }),
+      'await-first-dom-report',
+    )
+
+    const canvasControlsLayer = editor.renderedDOM.getByTestId(CanvasControlsContainerID)
+    const div = editor.renderedDOM.getByTestId(DivTestId)
+    const divBounds = div.getBoundingClientRect()
+
+    mouseClickAtPoint(canvasControlsLayer, {
+      x: divBounds.x + 5,
+      y: divBounds.y + 5,
+    })
+
+    const gapControlHandle = editor.renderedDOM.getByTestId(FlexGapControlHandleTestId)
+    const gapControlBounds = gapControlHandle.getBoundingClientRect()
+
+    const center = {
+      x: Math.floor(gapControlBounds.x + gapControlBounds.width / 2),
+      y: Math.floor(gapControlBounds.y + gapControlBounds.height / 2),
+    }
+
+    const endPoint = {
+      x: Math.floor(gapControlBounds.x + gapControlBounds.width / 2) + 11,
+      y: Math.floor(gapControlBounds.y + gapControlBounds.height / 2),
+    }
+
+    mouseDragFromPointToPoint(gapControlHandle, center, endPoint)
+    await editor.getDispatchFollowUpActionsFinished()
+
+    expect(getPrintedUiJsCode(editor.getEditorState())).toEqual(
+      testCodeWithGap({ gap: `gap: '53px',`, flexDirection: 'row' }),
```

Comment:
Minor, but the starting (numeric) value and drag delta could be constants, meaning this expected value could then be an expression to clearly indicate that it is exactly the result of applying that delta to the starting value.

Reply:
OK, so it would be something like
```
const initialGap = 42
const dragDelta = 11
const expectedGap = initialGap + dragDelta
// ... later in the expect
testCodeWithGap({ gap: `gap: '${expectedGap}px',`, flexDirection: 'row' })
```?

Reply:
Sure, or
```testCodeWithGap({ gap: `gap: '${initialGap + dragDelta}px',`, flexDirection: 'row' })```

Reply:
done!